### PR TITLE
Allow passing in a `DataTransferFile` to workspace edit

### LINKED ISSUE #175800: Avoid extra data copies when using copy/paste or drop APIs with workspace edits
From 169137

Using the copy or drop APIs, it's common for extensions to write the copied file back into the workspace with a workspace edit. However this currently result in a lot of data being copied back and forth to the extension host:

1. Copy event. We read the `File` from the event into a `UInt8Array`
2. Copy api is triggered
3. Extension host, we read the file data. This requires transferring that data from the main process to the ext host
4. Extension host, we create a new workspace edit using the file data
5. We transfer the workspace edit back to the main process

This means that we end up transferring a potentially large `Uint8Array` to the ext host and back for no real reason since the data is never changed. It would be better to keep all this in the main thread

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #175800

Allows you to pass a file `DataTransferFile` to `WorkspaceEdit.createFile`. This lets us avoid transferring the file data back and forth to the extension host, and also avoids base64 encoding and decoding. This significantly improving performance for large files

### CODE REVIEW

**Comment by mjbvz** [comment]
@jrieken Let me know what you think about this approach. The PR still needs some cleanup and any API changes would have to go through API review

**Comment by jrieken** [comment]
Isn't `DataTransferItem` is web-only API, meaning we pull in types that aren't vanilla JavaScript?

**Comment by mjbvz** [comment]
@jrieken This is actually our own [DataTransfer type](https://github.com/microsoft/vscode/blob/4d6e189ee1d17cee66d5fd2c94bc3265c83307b2/src/vscode-dts/vscode.d.ts#L10440) instead of the standard DOM one

File: src/vscode-dts/vscode.d.ts [hunk 1]
```diff
@@ -3710,7 +3710,18 @@ declare module 'vscode' {
...
+			readonly ignoreIfExists?: boolean;
+
+			/**
+			 * The initial contents of the new file.
+			 *
+			 * If creating a file from a {@link DocumentDropEditProvider drop operation}, you can
+			 * pass in a {@link DataTransferFile} to improve performance by avoiding extra data copying.
+			 */
+			readonly contents?: Uint8Array | DataTransferFile;
+		}, metadata?: WorkspaceEditEntryMetadata): void;
```
Comment (rebornix) [inline src/vscode-dts/vscode.d.ts:3724]: do we want to go through proposed dts first to test it out or it's already discussed in the sync meeting?

**Review by rebornix — APPROVED** [review]

**Review by joyceerhl — APPROVED** [review]

**Pull request merged**